code/datums/greyscale/layer.dm

```diff
@@ -123,11 +123,15 @@
 					continue
 				icon_file_colors.Add(pixel)
 
-/datum/greyscale_layer/hyperscale/InternalGenerate(list/colors, list/render_steps)
+/datum/greyscale_layer/hyperscale/InternalGenerate(list/colors, list/render_steps, datum/greyscale_config/parent)
 	. = ..()
 	var/icon/new_icon = icon(icon)
-	if(length(icon_file_colors) > length(colors))
+	if((length(icon_file_colors) > length(colors) && !CHECK_BITFIELD(parent.greyscale_flags, HYPERSCALE_ALLOW_GREYSCALE)))
 		CRASH("[src] set to Hyperscale, expected [length(icon_file_colors)], got [length(colors)].")
+	if(CHECK_BITFIELD(parent.greyscale_flags, HYPERSCALE_ALLOW_GREYSCALE) && (length(colors) == 1))
+		new_icon.Blend(colors[1], ICON_MULTIPLY)
+		return new_icon
+
```

Comment:
This flag allows the hyperscale layer to blend with only one color. It checks if colors == 1 because if it didnt, it would always just blend with the first color, even when passed a hyperscale palette
This flag is supposed to allow single color input on top of the hyperscale input